docker_cli/image.py

```diff
@@ -0,0 +1,430 @@
+            built. Defaults to "host"
+        no-cache: bool
+            A boolean designating whether or not the docker build should use
+            the cache
+
+        Returns
+        -------
+        Image
+            An Image class instance that references a Docker image built from
+            the Dockerfile at context/dockerfile
+
+        Raises
+        -------
+        CalledProcessError
+            Raised if the docker build command fails. Holds the value returned
+            by the docker build command in its returncode attribute, if not 0.
+        """
+        ...
+
+    @overload
+    @classmethod
+    def build(
+        cls: Type[Self],
+        context: os.PathLike[str],
+        tag: str,
+        *,
+        dockerfile_string: str,
+        output_file: Optional[io.TextIOBase],
+        network: str,
+        no_cache: bool
+    ):
+        """Build a Dockerfile at the given path with the given name, then
+        return the associated Image instance.
+
+        Parameters
+        ----------
+        context : str
+            The path to the file containing the Dockerfile
+        tag : str
+            A name for the image
+        dockerfile_string : str
+            A Dockerfile-formatted string containing the information necessary
+            to build the desired image.
+        output_file : io.TextIOBase, optional
+            A file-like object that the stderr output of the docker build
+            program will be written to. If None, the method will not output to
+            a file. Defaults to None.
+        network : str
+            The name of the network associated with the docker image to be
+            built. Defaults to "host"
+        no-cache: bool
+            A boolean designating whether or not the docker build should use
+            the cache
+
+        Returns
+        -------
+        Image
+            An Image class instance that references a Docker image built from
+            the Dockerfile-formatted string at context/dockerfile_loc
+
+        Raises
+        -------
+        CalledProcessError
+            Raised if the docker build command fails. Holds the value returned
+            by the docker build command in its returncode attribute, if not 0.
+        """
+        ...
+
+    @classmethod
+    def build(
+        cls,
+        context,
+        tag,
+        *,
+        dockerfile=None,
+        dockerfile_string=None,
+        output_file=None,
+        network="host",
+        no_cache=True
+    ):
+        if dockerfile is not None:
+            dockerfile_build = True
+        elif dockerfile_string is not None:
+            dockerfile_build = False
+        else:
+            return ValueError("Image.build: either dockerfile or " +
+                              "dockerfile_string must be defined.")
+
+        if output_file is not None:
+            run_stdout = output_file
+        else:
+            run_stdout = PIPE
```

Comment:
I don't think this part is necessary -- I think we could pass the `output_file` argument directly to `subprocess.run()` like this:

```python
subprocess.run(..., stdout=output_file, ...)
```

`None` is a valid value for `stdout` and I think it has the correct default behavior that we want, so I don't think we need to replace it with `PIPE`.

>stdin, stdout and stderr specify the executed program’s standard input, standard output and standard error file handles, respectively. Valid values are [PIPE](https://docs.python.org/3/library/subprocess.html#subprocess.PIPE), [DEVNULL](https://docs.python.org/3/library/subprocess.html#subprocess.DEVNULL), an existing file descriptor (a positive integer), an existing [file object](https://docs.python.org/3/glossary.html#term-file-object) with a valid file descriptor, and None. [PIPE](https://docs.python.org/3/library/subprocess.html#subprocess.PIPE) indicates that a new pipe to the child should be created. [DEVNULL](https://docs.python.org/3/library/subprocess.html#subprocess.DEVNULL) indicates that the special file [os.devnull](https://docs.python.org/3/library/os.html#os.devnull) will be used. With the default settings of None, no redirection will occur; the child’s file handles will be inherited from the parent. Additionally, stderr can be [STDOUT](https://docs.python.org/3/library/subprocess.html#subprocess.STDOUT), which indicates that the stderr data from the applications should be captured into the same file handle as for stdout.

https://docs.python.org/3/library/subprocess.html#subprocess.Popen